lib/spack/spack/solver/asp.py

```diff
@@ -446,9 +448,18 @@ def pkg_version_rules(self, pkg):
             #    c) Numeric or string comparison
             v)
 
-        most_to_least_preferred = sorted(
-            self.possible_versions[pkg.name], key=keyfn, reverse=True
-        )
+        # Compute which versions appear only in packages.yaml
+        from_externals = self.versions_from_externals[pkg.name]
+        from_package_py = self.versions_in_package_py[pkg.name]
+        only_in_packages_yaml = from_externals - from_package_py
```

Comment:
```suggestion
        only_from_externals = from_externals - from_package_py
```

Reply:
Done, missed this variable on a first pass.